### PR TITLE
Added 'pearson' to methods list in pandas/core/nanops.py

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [x] ref https://github.com/pandas-dev/pandas/pull/30461#discussion_r362323260
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: ci/deps/azure-macos-36.yaml [hunk 1]
```diff
@@ -26,6 +26,7 @@ dependencies:
   - pytables
   - python-dateutil==2.6.1
   - pytz
+  - scipy>=1.1
```
Comment (jreback) [inline ci/deps/azure-macos-36.yaml]: hmm, we *should* be skipping tests if the deps are not installed, which test is the issue?

**Review by jreback — CHANGES_REQUESTED** [review]

File: ci/deps/azure-macos-36.yaml (hunk 1 above)
Comment (MomIsBestFriend) [reply ci/deps/azure-macos-36.yaml]: > hmm, we _should_ be skipping tests if the deps are not installed, which test is the issue?

```
pandas.tests.frame.methods.test_cov_corr.TestDataFrameCorrWith
test_moments_rolling.TestRollingMomentsConsistency
pandas.tests.frame.methods.test_cov_corr.TestDataFrameCorrWith
pandas.tests.test_nanops.TestnanopsDataFrame
pandas.tests.test_nanops.TestnanopsDataFrame
pandas.tests.frame.methods.test_cov_corr.TestDataFrameCorrWith
pandas.tests.series.test_timeseries.TestTimeSeries
```

For the macOS build, at least.

Comment (jreback) [reply ci/deps/azure-macos-36.yaml]: then these should have a skip in place

File: pandas/core/nanops.py [hunk 2]
```diff
@@ -1241,7 +1241,7 @@ def nancorr(a, b, method="pearson", min_periods=None):
 
 
 def get_corr_func(method):
-    if method in ["kendall", "spearman"]:
+    if method in ["kendall", "spearman", "pearson"]:
```
Comment (jreback) [inline pandas/core/nanops.py]: actrually, why is this necessary at all? this just uses numpy and not scipy

Comment (MomIsBestFriend) [reply pandas/core/nanops.py]: You asked me to:)

ref - https://github.com/pandas-dev/pandas/pull/30461#discussion_r361339781

Comment (jreback) [reply pandas/core/nanops.py]: ahh right, but this is not actually the correct check, instead do like
```
if method in ['kendall', 'spearman']:
   .....
elif method in ['pearson']:
    pass
elif callable(method):
   ....
else:
     raise ValueError("....in valid method passed')
```

Comment (MomIsBestFriend) [reply pandas/core/nanops.py]:
```python
raise ValueError(f"Unrecognized method '{method}'")
```

Sounds good? (probably not, my English is not so good)

And also, should I remove all the "Skip" decorators Iv'e added in https://github.com/pandas-dev/pandas/pull/30603/commits/ed6f9c951be18edfdd90a659ae3f833ba8dcbbe0 ?

Comment (jreback) [reply pandas/core/nanops.py]: sure & yes remove the skips

File: pandas/core/nanops.py [hunk 3]
```diff
@@ -1243,8 +1243,14 @@ def nancorr(a, b, method="pearson", min_periods=None):
 def get_corr_func(method):
     if method in ["kendall", "spearman"]:
         from scipy.stats import kendalltau, spearmanr
+    elif method in ["pearson"]:
+        pass
     elif callable(method):
         return method
+    else:
+        raise ValueError(
+            f"Unkown method '{method}', expected one of 'kendall', 'spearman'"
```
Comment (jreback) [inline pandas/core/nanops.py]: can you add a test that hits this (likely just calling .corr(method='foo') would do it

**Comment by MomIsBestFriend** [comment]
Closed&Reopened because the build failed during compile time.

Checking if that was a false positive.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @MomIsBestFriend
